pkg/types/entries.go

```diff
@@ -121,12 +121,13 @@ func CanonicalizeEntry(ctx context.Context, entry EntryImpl) ([]byte, error) {
 // ArtifactProperties provide a consistent struct for passing values from
 // CLI flags to the type+version specific CreateProposeEntry() methods
 type ArtifactProperties struct {
-	ArtifactPath   *url.URL
-	ArtifactHash   string
-	ArtifactBytes  []byte
-	SignaturePath  *url.URL
-	SignatureBytes []byte
-	PublicKeyPath  *url.URL
-	PublicKeyBytes []byte
-	PKIFormat      string
+	AdditionalAuthenticatedData string
```

Comment:
should this get changed to `[]byte` and leave it up to the caller to decode the input correctly?

Reply:
Yes, that makes sense! So basically moving the base64 decoding into the cli part instead of `CreteFromArtifactProperties`. I'll give it a shot to see if I can make it happen.